src/psyclone/parse/module_info.py

```diff
@@ -210,3 +309,88 @@ def get_used_symbols_from_modules(self):
+        on the fparser tree (see get_parse_tree), and the information is
+        cached. If the PSyIR must be modified, it needs to be copied,
+        otherwise the modified tree will be returned from the cache in the
+        future.
+        If the conversion to PSyIR fails, a dummy FileContainer with an
+        empty Container (module) is returned, which avoids additional error
+        handling in many other subroutines.
+        #TODO 2120: This should be revisited when improving on the error
+        handling.
+
+        :param routine_name: optional the name of a routine.
+        :type routine_name: Optional[str]
+
+        :returns: PSyIR representing this module.
+        :rtype: List[:py:class:`psyclone.psyir.nodes.Node`]
+
+        '''
+        if self._psyir is None:
+            try:
+                self._psyir = \
+                    self._processor.generate_psyir(self.get_parse_tree())
+            except (KeyError, SymbolError, InternalError, FortranSyntaxError):
+                # Create a dummy FileContainer with a dummy module. This avoids
+                # additional error handling in other subroutines, since they
+                # will all return 'no information', whatever you ask for
+                self._psyir = FileContainer(os.path.basename(self._filename))
+                module = Container("invalid-module")
+                self._psyir.children.append(module)
+            # Store the PSyIR of all routines:
+            self._psyir_of_routines = {}
+            for routine in self._psyir.walk(Routine):
+                self._psyir_of_routines[routine.name.lower()] = routine
+
+        if routine_name is not None:
+            return self._psyir_of_routines[routine_name.lower()]
+        return self._psyir
+
+    # ------------------------------------------------------------------------
+    def resolve_routine(self, routine_name):
+        '''This function returns a list of function names that might be
+        actually called when the routine `name` is called. In most cases
+        this is exactly name, but in case of a generic subroutine the
+        name might change. For now (since we cannot resolve generic
+        interfaces yet), we return the list of all possible functions that
+        might be called.
+
+        :param str routine_name: the name of the routine to resolve
+
+        :returns: list of routine name(s) that could be called.
+        :rtype: List[str]
+
+        '''
+        if self._psyir_of_routines is None:
+            self.get_psyir()
+        routine_name = routine_name.lower()
+        if routine_name not in self._generic_interfaces:
+            return [routine_name]
+
+        # If a generic interface name is queried, return a copy
+        # of all possible routine names that might be called:
+        return self._generic_interfaces[routine_name][:]
+
+    # ------------------------------------------------------------------------
+    def get_symbol(self, name):
```

Comment:
Neither am I .... it's actually not used. I'd guess it comes with one of the next PRs, but I have removed it, and will remove its usage wherever it might be :)